src/count_slots.py

```diff
@@ -0,0 +1,231 @@
+from collections import defaultdict
+import concurrent.futures
+from datetime import date
+import functools
+import gzip
+import json
+import os
+from pathlib import Path
+import re
+
+
+FILE_DATE_PATTERN = re.compile(r'-(\d\d\d\d-\d\d-\d\d)\.')
+
+
+def read_json_lines(filepath, compressed=None):
+    if compressed is None:
+        compressed = str(filepath).endswith('.gz')
+
+    context = gzip.open(filepath, 'rt') if compressed else open(filepath)
+    with context as f:
+        for line in f:
+            if line and line != '\n':
+                yield json.loads(line)
+
+
+def deduplicate_locations(data, id_file):
+    """
+    Identify duplicate locations using a recent dump of the ``external_ids``
+    table and combine slot counts from locations that are duplicates. (This is
+    important because historical logs will contain reports for different
+    location IDs that we later learned were duplicates.)
+    """
+    clean = {}
+    lookup = {}
+    for row in read_json_lines(id_file):
+        location_id = row['provider_location_id']
+        if not (location_id in lookup):
+            clean[location_id] = defaultdict(lambda: 0)
```

Comment:
supernit, but more conventional would be `defaultdict(int)`. Or maybe `collections.Counter()` instead?

Reply:
Ah, yeah, I guess this is one of those unusual things I learned from someone way back, on the theory that it reads more clearly. But `defaultdict(int)` is so common it probably doesn't matter.

I always find `Counter` to be awkward for this kind of use because it has no `add(key)` or `increment(key)` method. Instead you have to `update({'key': 1})` or `+ {'key': 1}`, which is verbose and requires unnecessary allocations (which are in fact pretty expensive over the amount of iterations this script makes).

Reply:
re: updates, you should be able to leave the rest of the code as is; `+=` does in place updates.